test/utils/test_struct.py

```diff
@@ -168,3 +168,40 @@ def test_cache_hash():
     assert a.__Point0cache_hash_cache == 123
     object.__setattr__(a, "__Point0cache_hash_cache", 1234)
     hash(a) == 1234
+
+
+@struct.dataclass
+class PointC:
+    x: float
+    y: float
+
+    @struct.property_cached(pytree_node=True)
+    def cached_node(self) -> int:
+        return self.x * self.y
+
+
+# test for pytree_node=True cached properties
+def test_cached_pytreenode_properties():
+    p = PointC(2.0, 3.0)
+    assert p.__cached_node_cache is struct.Uninitialized
+    assert p.cached_node == 2.0 * 3.0
+    assert p.__cached_node_cache == 6.0
+
+    p = p.replace(y=3.0)
+    assert p.__cached_node_cache is struct.Uninitialized
+    assert p.cached_node == 2.0 * 3.0
+    assert p.__cached_node_cache == 6.0
+
+    @jax.jit
+    def compute(x: PointC):
+        return x.cached_node * 3, x
+
+    p = PointC(2.0, 3.0)
+    res, p2 = compute(p)
+    assert p.__cached_node_cache is struct.Uninitialized
+    assert res == 6.0 * 3
+    assert p2.__cached_node_cache == 6.0
+
+    res, p3 = compute(p2)
+    assert res == 6.0 * 3
+    assert p3.__cached_node_cache == 6.0
```

Comment:
Could you maybe add a comment what exactly this tests? `p`, `p2`, `p3` are the same object (since `compute` just returns `x` as second value), right?

Reply:
that's testing that everything works under jit (because jit implicitly unpacks-then-repacks objects)

Reply:
i added a comment.